### DISCUSSION TITLE
FB ads with random number

### DISCUSSION THREAD

**Discussion by IgnoreBox** [uBlockOrigin/uAssets#15999] [Ideas]
i found this
```adb
facebook.com##div.x1lliihq:nth-of-type(1)
facebook.com##div.x1lliihq:nth-of-type(2)
facebook.com##div.x1lliihq:nth-of-type(3)
facebook.com##div.x1lliihq:nth-of-type(4)
facebook.com##div.x1lliihq:nth-of-type(5)
facebook.com##div.x1lliihq:nth-of-type(6)
facebook.com##div.x1lliihq:nth-of-type(7)
facebook.com##div.x1lliihq:nth-of-type(8)
facebook.com##div.x1lliihq:nth-of-type(9)
facebook.com##div.x1lliihq:nth-of-type(10)
facebook.com##div.x1lliihq:nth-of-type(11)
facebook.com##div.x1lliihq:nth-of-type(12)
facebook.com##div.x1lliihq:nth-of-type(13)
facebook.com##div.x1lliihq:nth-of-type(14)
facebook.com##div.x1lliihq:nth-of-type(15)
facebook.com##div.x1lliihq:nth-of-type(16)
facebook.com##div.x1lliihq:nth-of-type(17)
facebook.com##div.x1lliihq:nth-of-type(18)
facebook.com##div.x1lliihq:nth-of-type(19)
facebook.com##div.x1lliihq:nth-of-type(20)
```

my idea was how to make facebook.com##div.x1lliihq:nth-of-type(*)

thankyou 

ps: im learning to create my own filter list

**Comment by mapx-** [comment]
you need only

`facebook.com##div.x1lliihq`

**Reply by IgnoreBox** [reply]
so if we found some random text behind the div or something... just delete the random thing ?

ok i will test it

**Reply by IgnoreBox** [reply]
not working :( but if i put full 1 line by line 1-20
some ads are disappear and if sponsored appear again it must higher than 20

**Reply by krystian3w** [reply]
```adb
facebook.com##div.x1lliihq:nth-of-type(-n+20)
```

**Reply by IgnoreBox** [reply]
> ```
> facebook.com##div.x1lliihq:nth-of-type(-n+20)
> ```

thankyou

**Comment by mapx-** [comment]
if you have

```
example.com###fixpartrandom1
example.com###fixpartrandom2
example.com###fixpartrandom3
```

you can use
`example.com##[id^="fixpart"]`

or if a class

```
example.com##.fixpartrandom1
example.com##.fixpartrandom2
example.com##.fixpartrandom3
```

you can use
`example.com##[class^="fixpart"]`

read more
https://help.adblockplus.org/hc/en-us/articles/360062733293#elemhide_basic

**Reply by IgnoreBox** [reply]
let me read and try 1st..... thank you before ..

**Comment by IgnoreBox** [comment]
some of answer here was working on my test website
but not working at facebook

i think i must let facebook ads..... they smarter than me :D

**Reply by krystian3w** [reply]
Maybe upgraded to use shadow root https://github.com/uBlockOrigin/uBlock-issues/discussions/2280

**Reply by IgnoreBox** [reply]
let me check it first

**Reply by IgnoreBox** [reply]
perhaps that is the solution.. but i cant understand :D since im a newbie.. and trying to start making my own filter
but... thankyou. i bookmarked that page

**Reply by krystian3w** [reply]
I mean why filter can fail, do not linked discussion with solution.

**Reply by IgnoreBox** [reply]
yes,, i understand that... because 'why' we can try 'how' right ?

**Comment by coccbolcker-pixel** [comment]
I'm posting this here because... I am not doing the reporting process. brother I'm a normal person who don't work on computers. I don't know what those computer related words mean. I know how to turn the thing off and on. I know how to install and delete programs. I can type. that's about the extent of my computer abilities. I'm not stupid, just didn't do computers when I was growing up. they were expensive back then. 

anyway- the problem. if you could call it that.

facebook is showing ads in the "sponsored" section again. I have complete coverage on all the time. I've cycled the coverage level and refreshed the page each time and they don't go away. 

it doesn't bother me beyond the fact facebook on chrome browser got around it again. just want to put the info out there. maybe someone if they want to, look into that? I can't be the only one with this odd issue.

**Reply by mapx-** [reply]
all facebook issues are managed here:
https://github.com/uBlockOrigin/uAssets/issues/3367
